File: src/main/java/com/github/kory33/signvote/manager/vote/UUIDToPlayerVotesCacheMap.java
package com.github.kory33.signvote.manager.vote;

import com.github.kory33.signvote.utils.collection.CachingMap;

import java.util.UUID;

/**
 * A map collection class that has {@link UUID} as keys and {@link VoteScoreToVotePointCacheMap} as values.
 * This map automatically creates empty value upon call of {@link #get(Object)} method where value is {@code null}
 *
 * @see com.github.kory33.signvote.utils.collection.CachingMap
 */
/*package-private*/ class UUIDToPlayerVotesCacheMap extends CachingMap<UUID, VoteScoreToVotePointCacheMap> {
    UUIDToPlayerVotesCacheMap() {
        super(VoteScoreToVotePointCacheMap::new);
    }
}

File: src/main/kotlin/com/github/kory33/signvote/manager/vote/UUIDToPlayerVotesCacheMap.kt
package com.github.kory33.signvote.manager.vote

import com.github.kory33.signvote.utils.collection.CachingMap

import java.util.UUID

/**
 * A map collection class that has [UUID] as keys and [VoteScoreToVotePointCacheMap] as values.
 * This map automatically creates empty value upon call of [.get] method where value is `null`

 * @see com.github.kory33.signvote.utils.collection.CachingMap
 */

internal class UUIDToPlayerVotesCacheMap :
        CachingMap<UUID, VoteScoreToVotePointCacheMap>({ VoteScoreToVotePointCacheMap() })
